Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1367,6 +1456,18 @@ private static void OnItemsSourcePropertyChanged(DependencyObject d, DependencyP
                 // can be set when the DataGrid is not part of the visual tree.
                 dataGrid._measured = false;
                 dataGrid.InvalidateMeasure();
+
+                // Determine if incremental loading should be used
+                if (dataGrid.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
```

Comment:
@MSRegisB If I understand you correctly, would this be suitable?

```
if (dataGrid.DataConnection.DataSource is ISupportIncrementalLoading incrementalDataSource)
{
	dataGrid._incrementalItemsSource = incrementalDataSource;
	dataGrid._loadingOperation.Cancel();
	dataGrid._loadingItems = false;
}
else if (dataGrid.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
{
	dataGrid._incrementalItemsSource = incrementalItemsSource;
	dataGrid._loadingOperation.Cancel();
	dataGrid._loadingItems = false;
}
else
{
	dataGrid._incrementalItemsSource = default(ISupportIncrementalLoading);
	dataGrid._loadingOperation.Cancel();
	dataGrid._loadingItems = false;
}
```

Reply:
Yes, that's the spirit. But I do have some feedback here:
I would prefer the incremental loading code to be part of the data-centric DataGridDataConnection class. What I mean is move _incrementalItemsSource and _loadingOperation into the DataGridDataConnection class, where _dataSource is declared.
And then the logic above can be placed in a new DataGridDataConnection::UpdateIncrementalItemsSource() method, which would be called at the end of the DataGridDataConnection::DataSource setter.
Your code above would be simplified to:

if (_dataSource is ISupportIncrementalLoading incrementalDataSource)
{
	_incrementalItemsSource = incrementalDataSource;
}
else if (_owner.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
{
	_incrementalItemsSource = incrementalItemsSource;
}
else
{
	_incrementalItemsSource = default(ISupportIncrementalLoading);
}

if (_loadingOperation != null)
{
	_loadingOperation.Cancel();
	_loadingOperation = null;
}

_loadingItems does not seem needed anymore (_loadingItems is equivalent to _loadingOperation != null ?).

Of course, "await _incrementalItemsSource.LoadMoreItemsAsync(numberOfRowsToLoad);" would also be moved into a new DataGridDataConnection::LoadMoreItems(uint count) method.

Thanks.

Reply:
@MSRegisB I don't think removing _loadingItems would work as it is used to ensure that two or more loading processes are not started at the same time. Unless of course, _loadingOperation becomes null upon completion.

Just a thought, perhaps we should even consider making  _loadingItems bindable so that it can be bound to a ProgressBar or similar?

If we wrap LoadMoreItemsAsync(uint) inside DataConnection, I guess we should do the same for HasMoreItems? 

Regarding DataGridDataConnection::LoadMoreItems, would you like it to be awaitable or just a normal void?